Mathlib/Algebra/BigOperators/Basic.lean

```diff
@@ -90,43 +90,160 @@ and we therefore choose the level `67`.
 
 In practice, this means that parentheses should be placed as follows:
 ```lean
-∑ k in K, (a k + b k) = ∑ k in K, a k + ∑ k in K, b k →
-  ∏ k in K, a k * b k = (∏ k in K, a k) * (∏ k in K, b k)
+∑ k ∈ K, (a k + b k) = ∑ k ∈ K, a k + ∑ k ∈ K, b k →
+  ∏ k ∈ K, a k * b k = (∏ k ∈ K, a k) * (∏ k ∈ K, b k)
 ```
 (Example taken from page 490 of Knuth's *Concrete Mathematics*.)
 -/
 
 -- TODO: Use scoped[NS], when implemented?
 namespace BigOperators
-open Std.ExtendedBinder
+open Std.ExtendedBinder Lean Meta
+
+/-- `finset% t` elaborates `t` as a `Finset`.
+If `t` is a `Set`, then inserts `Set.toFinset`.
+Does not make use of the expected type; useful for big operators over finsets.
+```
+#check finset% Finset.range 2 -- Finset Nat
+#check finset% (Set.univ : Set Bool) -- Finset Bool
+```
+-/
+elab (name := finsetStx) "finset% " t:term : term => do
+  let u ← mkFreshLevelMVar
+  let ty ← mkFreshExprMVar (mkSort (.succ u))
+  let x ← Elab.Term.elabTerm t (mkApp (.const ``Finset [u]) ty)
+  let xty ← whnfR (← inferType x)
+  if xty.isAppOfArity ``Set 1 then
+    Elab.Term.elabAppArgs (.const ``Set.toFinset [u]) #[] #[.expr x] none false false
+  else
+    return x
+
+open Lean.Elab.Term.Quotation in
+/-- `quot_precheck` for the `finset%` syntax. -/
+@[quot_precheck BigOperators.finsetStx] def precheckFinsetStx : Precheck
+  | `(finset% $t) => precheck t
+  | _ => Elab.throwUnsupportedSyntax
+
+-- TODO: contribute this modification back to `extBinder`
```

Comment:
The changes to the `extBinder` syntax are in https://github.com/leanprover/std4/pull/234/files and in the meantime it's given as new syntax as `bigOpBinder` here.